statefun-e2e-tests/statefun-smoke-e2e-java/statefun-smoke-e2e-fn/src/main/java/org/apache/flink/statefun/e2e/smoke/CommandInterpreter.java

```diff
@@ -0,0 +1,162 @@
+      Command::toByteArray,
+      bytes -> Command.parser().parseFrom(bytes));
+
+  static final Type<Commands> COMMANDS_TYPE = SimpleType.simpleImmutableTypeFrom(
+      TypeName.typeNameOf("statefun.smoke.e2e.types", "commands"),
+      Commands::toByteArray,
+      bytes -> Commands.parser().parseFrom(bytes));
+
+  static final Type<SourceCommand> SOURCE_COMMAND_TYPE = SimpleType.simpleImmutableTypeFrom(
+      TypeName.typeNameOf("statefun.smoke.e2e.types", "source-command"),
+      SourceCommand::toByteArray,
+      bytes -> SourceCommand.parser().parseFrom(bytes));
+
+  static final Type<VerificationResult> VERIFICATION_RESULT_TYPE = SimpleType.simpleImmutableTypeFrom(
+      TypeName.typeNameOf("statefun.smoke.e2e.types", "verification-result"),
+      VerificationResult::toByteArray,
+      bytes -> VerificationResult.parser().parseFrom(bytes));
+
+  static final TypeName DISCARD_EGRESS_TYPENAME =
+      TypeName.typeNameOf("statefun.smoke.e2e.types", "discard-egress");
+
+  static final TypeName VERIFICATION_EGRESS_TYPENAME =
+      TypeName.typeNameOf("statefun.smoke.e2e.types", "verification-result-egress");
+
+  public CommandInterpreter(Ids ids) {
+    this.ids = Objects.requireNonNull(ids);
+  }
+
+  public void interpret(ValueSpec<Long> state, Context context, Message message) {
+    if (message.is(SOURCE_COMMAND_TYPE)) {
+      interpret(state, context, message.as(SOURCE_COMMAND_TYPE).getCommands());
+    } else if (message.is(COMMANDS_TYPE)) {
+      interpret(state, context, message.as(COMMANDS_TYPE));
+    } else {
+      throw new IllegalArgumentException("Unrecognized message type!");
+    }
+  }
+
+  private void interpret(ValueSpec<Long> state, Context context, Commands cmds) {
+    for (Command cmd : cmds.getCommandList()) {
+      if (cmd.hasIncrement()) {
+        modifyState(state, context, cmd.getIncrement());
+      } else if (cmd.hasSend()) {
+        send(state, context, cmd.getSend());
+      } else if (cmd.hasSendAfter()) {
+        sendAfter(state, context, cmd.getSendAfter());
+      } else if (cmd.hasSendEgress()) {
+        sendEgress(state, context, cmd.getSendEgress());
+      } else if (cmd.hasVerify()) {
+        verify(state, context, cmd.getVerify());
+      }
+    }
+  }
+
+  private void verify(
+      ValueSpec<Long> state,
+      @SuppressWarnings("unused") Context context,
+      Command.Verify verify) {
+    AddressScopedStorage storage = context.storage();
+    Address targetAddress = new Address(Fn.TYPENAME, context.self().id());
+    long actual = storage.get(state).orElse(0L);
+    long expected = verify.getExpected();
+    VerificationResult verificationResult =
+        VerificationResult.newBuilder()
+            .setId(Integer.parseInt(context.self().id()))
+            .setActual(actual)
+            .setExpected(expected)
+            .build();
+    Message message =
+        EgressMessageBuilder.forEgress(VERIFICATION_EGRESS_TYPENAME)
```

Comment:
Send verification result to verification egress, will be binded with SocketSink and forwarded to "SimpleProtobufServer" for verification.